tests/unit/test_network_connections.py

```diff
@@ -3670,6 +3707,95 @@ def test_path_to_glib_bytes(self):
         self.assertEqual(result.get_data(), b"file:///my/test/path\x00")
 
 
+class TestValidatorMatch(unittest.TestCase):
+    def setUp(self):
+        self.test_profile = {
+            "name": "test",
+            "type": "ethernet",
+            "ip": {
+                "dhcp4": False,
+                "address": "192.168.245.7/24",
+            },
+            "match": None,
+        }
+        self.validator = network_lsr.argument_validator.ArgValidator_DictConnection()
+
+    def test_match_path_empty_list(self):
+        """
+        Test that 'match.path' setting can be defined as None, [], [""], [None] or ""
+        and such a setting will be normalized into []
+        """
+        for disabled_path in [None, [], [""], [None], ""]:
+            self.test_profile["match"] = {"path": disabled_path}
+            result = self.validator.validate(self.test_profile)
+            self.assertEqual(result["match"], {"path": []})
+
+    def test_match_path_valid_setting(self):
+        """
+        Test that values like ["&"] and ["pci-0000:00:03.0"] are valid values for
+        'match.path' setting
+        """
+        self.test_profile["match"] = {"path": ["&"]}
+        result1 = self.validator.validate(self.test_profile)
+        self.assertEqual(result1["match"], {"path": ["&"]})
```

Comment:
@thom311 following up from our discussion recently, this would mean that the profile will only match interfaces with empty PCI paths, which should not happen for ethernet/infiniband, should it?

Reply:
it probably would match only with devices that have no PCI path. Which doesn't make much sense...

Reply:
Ok, I guess the same holds true for `"|"`, so @liangwen12year it would be good to reject those values, too.

Reply:
I rejected the value ["|"], ["&"] in argvalidatorList.
```yaml
        if self.is_match_path:
            # 'match.path' setting ["|"], ["&"] will only match the devices that have
            # no PCI path, it is impossible for ethernet/infiniband, so raise a error
            if result == ["|"] or result == ["&"]:
                raise ValidationError(
                    name,
                    "value '%s' is not a valid 'match.path' setting, after "
                    "normalization, '%s' will only match the devices that have no PCI "
                    "path" % (value_before_norm, result),
                )
```